validator/client/src/test-support/java/tech/pegasys/artemis/validator/client/AttestationGenerator.java

```diff
@@ -0,0 +1,124 @@
+
+  public AttestationGenerator(final List<BLSKeyPair> validatorKeys) {
+    this.validatorKeys = validatorKeys;
+  }
+
+  public Attestation validAttestation(final ChainStorageClient storageClient) {
+    final Bytes32 bestBlockRoot = storageClient.getBestBlockRoot();
+    BeaconBlock block = storageClient.getStore().getBlock(bestBlockRoot);
+    BeaconState state = storageClient.getStore().getBlockState(bestBlockRoot);
+    return createAttestation(block, state, true);
+  }
+
+  public Attestation attestationWithInvalidSignature(final ChainStorageClient storageClient) {
+    final Bytes32 bestBlockRoot = storageClient.getBestBlockRoot();
+    BeaconBlock block = storageClient.getStore().getBlock(bestBlockRoot);
+    BeaconState state = storageClient.getStore().getBlockState(bestBlockRoot);
+    return createAttestation(block, state, false);
+  }
+
+  private Attestation createAttestation(
+      final BeaconBlock block, final BeaconState state, final boolean withValidSignature) {
+    final UnsignedLong epoch = compute_epoch_at_slot(state.getSlot());
+    HashMap<UnsignedLong, List<Triple<List<Integer>, UnsignedLong, Integer>>> committeeAssignments =
+        new HashMap<>();
+    int validatorIndex;
+    for (validatorIndex = 0; validatorIndex < validatorKeys.size(); validatorIndex++) {
+      final Optional<CommitteeAssignment> maybeAssignment =
+          ValidatorClientUtil.get_committee_assignment(state, epoch, validatorIndex);
+      if (maybeAssignment.isPresent()) {
+        UnsignedLong slot = maybeAssignment.get().getSlot();
+        final Triple<List<Integer>, UnsignedLong, Integer> committeeAssignment =
+            new MutableTriple<>(
+                maybeAssignment.get().getCommittee(), // List of validator indices in this committee
+                maybeAssignment.get().getCommitteeIndex(), // Assigned shard
+                validatorIndex // The index of the current validator
+                );
+        committeeAssignments.put(slot, List.of(committeeAssignment));
+        break;
+      }
+    }
+    if (committeeAssignments.isEmpty()) {
+      throw new IllegalStateException("Unable to find committee assignment among validators");
+    }
+
+    final UnsignedLong slot = committeeAssignments.keySet().iterator().next();
+    Triple<BLSPublicKey, Integer, Committee> attesterInfo =
+        AttestationUtil.getAttesterInformation(state, committeeAssignments, slot).get(0);
+    AttestationData genericAttestationData =
+        AttestationUtil.getGenericAttestationData(state, block, slot);
+
+    final BLSKeyPair validatorKeyPair =
+        withValidSignature ? validatorKeys.get(validatorIndex) : randomKeyPair;
+    return createAttestation(
+        state,
+        validatorKeyPair,
+        attesterInfo.getMiddle(),
+        attesterInfo.getRight(),
+        genericAttestationData);
+  }
+
+  private Attestation createAttestation(
+      BeaconState state,
+      BLSKeyPair attesterKeyPair,
```

Comment:
I'm a bit wary of having functions specifically for testing since we have very similar ones that are not being tested in production. Can we not share those functions? so that we know they're tested?

Reply:
👍Added this logic from master while we had the 0.9.0 update ongoing, so I tried to avoid modifying the core logic too much.  But I'll add a ticket to look into deduplicating this more aggressively.

Reply:
Ticket: https://pegasys1.atlassian.net/browse/BC-115